src/storage_types.rs

```diff
@@ -248,3 +248,18 @@ storage_type_types! {
     storage_type_f32!("f32", f32, f32);
     storage_type_f64!("f64", f64, f64);
 }
+
+macro_rules! impl_const_zero {
+    () => {};
+
+    ($type:ident $($rest:tt)*) => {
+        impl $crate::ConstZero for $type {
+            #[allow(trivial_numeric_casts)]
+            const ZERO: Self = 0 as Self;
+        }
+
+        impl_const_zero!($($rest)*);
+    };
+}
+
+impl_const_zero!(usize u8 u16 u32 u128 isize i8 i16 i32 i64 i128 f32 f64);
```

Comment:
Any reason not to use `storage_types!`?

```suggestion

storage_types! {
    types: PrimInt, Float;

    impl $crate::ConstZero for V {
        #[allow(trivial_numeric_casts)]
        const ZERO: Self = 0 as Self;
    }
}
```

Reply:
None at all. I have to admit to not understanding the storage types macro sufficiently to produce the above suggestion on my own.